android/modules/platform/src/java/ti/modules/titanium/platform/PlatformModule.java

```diff
@@ -337,6 +357,83 @@ public String getRuntime()
 		return KrollRuntime.getInstance().getRuntimeName();
 	}
 
+	// clang-format off
+	@Kroll.method
+	@Kroll.getProperty
+	public double getUptime()
+	// clang-format on
+	{
+		return SystemClock.uptimeMillis() / 1000.0;
+	}
+
+	// clang-format off
+	@Kroll.method
+	public Object[] cpus()
+	// clang-format on
+	{
+		List<Processor> processors = getProcessors();
+		List<KrollDict> result = new ArrayList<KrollDict>(processors.size());
+		for (Processor p : processors) {
+			result.add(p.toKrollDict());
+		}
+		return result.toArray();
+	}
+
+	private synchronized List<Processor> getProcessors()
+	{
+		if (this.processors != null) {
+			return this.processors;
+		}
+		final int processorCount = getProcessorCount();
+		processors = new ArrayList<Processor>(processorCount);
+
+		// Now read in their details
+		BufferedReader in = null;
+		List<Map> groups = new ArrayList<Map>();
+		Map<String, String> current = new HashMap<String, String>();
+		try {
+			String[] args = { "/system/bin/cat", "/proc/cpuinfo" };
```

Comment:
@sgtcoolguy, are you sure reading Android `/proc/cpuinfo` is reliable? Our APS library used to do this, and because of this, the **majority** of Android devices have been reported to our analytics server with an "unknown" architecture. Please see the following ticket. You can see our test results with different devices too.
https://jira.appcelerator.org/browse/TIMOB-25878

For APS, it could be because it made a bad assumption about a "Processor" field always existing. But regardless, we switched over to using a public Android API for obtaining CPU architecture which is definitely reliable. I see your code is scanning for "processor" (lowercase), so, you might get bit by this too.